data/yam/agama/auto/template.jsonnet

```diff
@@ -0,0 +1,20 @@
+local base_lib = import 'lib/base.libsonnet';
+local scripts_pre_lib = import 'lib/scripts_pre.libsonnet';
+local scripts_post_lib = import 'lib/scripts_post.libsonnet';
+local storage_lib = import 'lib/storage.libsonnet';
+
+function(bootloader=false, user=true, root=true, storage='', product='', scripts_pre='',
+  scripts_post='', encrypted=false) {
+  [if product != '' then 'product']: { id: product },
+  [if bootloader == true then 'bootloader']: base_lib['bootloader'],
+  [if user == true then 'user']: base_lib['user'],
+  [if root == true then 'root']: base_lib['root'],
+  [if scripts_pre != '' || scripts_post != '' then 'scripts']: {
+    [if scripts_pre != '' then 'pre']: [ scripts_pre_lib[x] for x in std.split(scripts_pre, ',') ],
+    [if scripts_post != '' then 'post']: [ scripts_post_lib[x] for x in std.split(scripts_post, ',') ],
+  },
+
+  [if storage == 'lvm' then 'storage']: storage_lib.lvm(encrypted),
+  [if storage == 'root_filesystem_ext4' then 'storage']: storage_lib['root_filesystem_ext4'],
+  [if storage == 'root_filesystem_xfs' then 'storage']: storage_lib['root_filesystem_xfs'],
+}
```

Comment:
could we  please order alphabetically all the keys (after the first `if`)  for easy searching.